### PR TITLE
`Development`: Move lecture text unit features into the lecture micro service 

### PR DESCRIPTION
**Pull request by MerlinMehmed**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] *This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [ ] *Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.
- [ ] * not related
### Motivation and Context

This PR moves functionality from Artemis into the Lecture microservice. It moves the lecture text unit resource and its tests.

### Description
The PR doesn't add new features, we just want to make sure the old ones are still working in the new architecture.

### Steps for Testing

Unfortunately, it is still not possible to test this PR on a test server. It should be tested locally. 
n order to do it, you need to run:
- JHipster Registry by running `docker-compose -f src/main/docker/jhipster-registry.yml up`
- Gateway application by running `./gradlew :gateway:bootRun`
- Lecture application by running `./gradlew :lecture:bootRun --args='--spring.profiles.active=artemis,dev'`
- Artemis as usual or i.e. `./gradlew bootRun --args='--spring.profiles.active=dev,bamboo,bitbucket,artemis,scheduling' -x webapp` to skip the client code build since it's not needed

Prerequisites:
- 1 Instructor
- 1 Student

0. Open [http://localhost:8089/](http://localhost:8089/)
1. Log in to Artemis
2. Navigate to Course Management -> Lectures -> Units
5. Create a new Text Unit
6. Open it, make sure you can access it 
7. Edit the created text unit and make sure the changes are saved
8. Check that you can access it as a student too

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### CODE REVIEW

**Review by b-fein — APPROVED** [review]
Code review: looks good.
Manual Tests: I had to change two settings in `src/main/resources/config/` to get Artemis to be registered in the registry. Afterwards the actual creation/editing/deletion of text units worked without issues. :+1: 

The two settings I had to change are:
- `application.yml`: `jhipster.registry.password: admin`
- `application-dev.yml`: `eureka.client.enabled: true` (the same parameter can be set in `application.yml` but I assume it is overwritten by the `dev` profile)

**Comment by MerlinMehmed** [comment]
Thank you for the review @b-fein. I have forgotten to commit the changes. I added them so that other people can test it without having change the configs

File: src/test/java/de/tum/in/www1/artemis/TextUnitIntegrationTest.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 import de.tum.in.www1.artemis.repository.UserRepository;
 import de.tum.in.www1.artemis.util.ModelFactory;
 
+@Deprecated // Moved to Lecture microservice. To be removed
```
Comment (julian-christl) [inline src/test/java/de/tum/in/www1/artemis/TextUnitIntegrationTest.java:19]: remove, see lecture unit PR

File: lecture/src/main/java/de/tum/in/www1/artemis/lecture/web/rest/TextUnitResource.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        }
+        Lecture lecture = lectureOptional.get();
+        if (lecture.getCourse() == null || (textUnit.getLecture() != null && !lecture.getId().equals(textUnit.getLecture().getId()))) {
+            throw new ConflictException();
+        }
+        authorizationCheckService.checkHasAtLeastRoleForLectureElseThrow(Role.EDITOR, lecture, null);
+
+        // persist lecture unit before lecture to prevent "null index column for collection" error
+        textUnit.setLecture(null);
+        textUnit = textUnitRepository.saveAndFlush(textUnit);
```
Comment (julian-christl) [inline lecture/src/main/java/de/tum/in/www1/artemis/lecture/web/rest/TextUnitResource.java:120]: fix codacy

**Review by julian-christl — CHANGES_REQUESTED** [review]

**Review by julian-christl — APPROVED** [review]
Code looks good

**Pull request merged**